qiskit/circuit/random/utils.py

```diff
@@ -11,16 +11,385 @@
 # that they have been altered from the originals.
 
 """Utility functions for generating random circuits."""
-
 import numpy as np
 
-from qiskit.circuit import ClassicalRegister, QuantumCircuit, CircuitInstruction
-from qiskit.circuit import Reset
+from qiskit.circuit import (
+    ClassicalRegister,
+    QuantumCircuit,
+    CircuitInstruction,
+)
+from qiskit.circuit import Reset, Delay
 from qiskit.circuit.library import standard_gates
+from qiskit.circuit.library.standard_gates import get_standard_gate_name_mapping
 from qiskit.circuit.exceptions import CircuitError
 from qiskit.quantum_info.operators.symplectic.clifford_circuits import _BASIS_1Q, _BASIS_2Q
 
 
+def _get_gates(n_qubits):
+    return [
+        (gate, gate.num_qubits, len(gate.params))
+        for gate in get_standard_gate_name_mapping().values()
+        if gate.num_qubits == n_qubits
+    ]
+
+
+def random_circuit_from_graph(
+    interaction_graph,
```

Comment:
I think it would be very nice from user experience to also allow a user to just send a list of tuples, say `[(0, 1, 1), (1, 2, 1), (2, 3, 2)]` and have the function handle itself the rustworkx construction (inferring the number of nodes automatically in this case, based on the edges and say filling up holes in node indices, if there are any).

Reply:
Just to adhere with the instruction in the issue, I decided to keep it as `PyDiGraph`
> "The [generation of random circuits](https://github.com/Qiskit/qiskit/blob/main/qiskit/circuit/random/utils.py) should be extended by the ability to specify a circuit connectivity or interaction graph G that the random graph should should adhere to."

The requested changes are in [eab2a4c](https://github.com/Qiskit/qiskit/pull/12474/commits/eab2a4c191839518ad554ab8bfcad23370405084)